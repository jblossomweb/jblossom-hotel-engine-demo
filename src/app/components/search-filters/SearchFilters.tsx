import React from 'react';
import { Tag, Select } from 'antd';
import { CodeOutlined } from '@ant-design/icons';
import languagesPopular from '../../data/languages-popular.json';
import languagesOther from '../../data/languages-other.json';
import * as Style from './SearchFilters.style';

export interface Props {
  loading?: boolean;
  filteredLanguages: string[];
  addFilteredLanguage: (language: string) => void;
  removeFilteredLanguage: (language: string) => void;
}

const SearchFilters: React.FC<Props> = ({
  loading,
  filteredLanguages,
  addFilteredLanguage,
  removeFilteredLanguage,
}) => (
  <Style.Wrapper>
    <Select
      disabled={loading}
      mode="multiple"
      style={{ width: '100%' }}
      placeholder={
        <>
          <CodeOutlined /> language
        </>
      }
      tagRender={({ value }) => (
        <Tag
          closable
          onClose={() => {
            removeFilteredLanguage(String(value));
          }}
        >
          <CodeOutlined /> {value}
        </Tag>
      )}
      value={filteredLanguages}
      onChange={(newValue) => {
        if (newValue.length > filteredLanguages.length) {
          newValue.forEach((language) => {
            if (!filteredLanguages.includes(language)) {
              addFilteredLanguage(language);
            }
          });
        }
      }}
    >
      <Select.OptGroup label="Popular">
        {languagesPopular.map((language) => (
          <Select.Option value={language} key={`popular-${language}`}>
            {language}
          </Select.Option>
        ))}
      </Select.OptGroup>
      <Select.OptGroup label="All">
        {languagesOther.map((language) => (
          <Select.Option value={language} key={`other-${language}`}>
            {language}
          </Select.Option>
        ))}
      </Select.OptGroup>
    </Select>
  </Style.Wrapper>
);

export default SearchFilters;
