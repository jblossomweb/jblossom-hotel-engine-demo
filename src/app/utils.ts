export const requestTimeout = (ms: number) =>
  new Promise((_resolve, reject) => {
    const wait = setTimeout(() => {
      clearTimeout(wait);
      reject(new Error('Request Timeout'));
    }, ms);
  });

export const trimWords = (words: string) =>
  words
    .split(' ')
    .filter((word) => word.length)
    .map((word) => word.trim())
    .join(' ');
