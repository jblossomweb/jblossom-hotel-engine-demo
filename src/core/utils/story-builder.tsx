/* eslint-disable @typescript-eslint/no-explicit-any */
import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import * as knobs from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';

import 'antd/dist/antd.css';
import Theme from '../../app/theme.style';

const { withKnobs } = knobs;

export interface KnobsInterface {
  text: (name: string, val: string) => string;
  number: (name: string, val: number) => number;
  date: (name: string, val: number) => number;
  color: (name: string, val: string) => string;
  boolean: (name: string, val: boolean) => boolean;
  object: (name: string, val: any) => any;
  select: (name: string, options: any, val: any) => any;
}

export const mockKnobs: KnobsInterface = {
  text: (name: string, val: string) => val,
  number: (name: string, val: number) => val,
  date: (name: string, val: number) => val,
  color: (name: string, val: string) => val,
  boolean: (name: string, val: boolean) => val,
  object: (name: string, val: object) => val,
  select: (name: string, options: any, val: any) => val,
};

export type Story = (...args: any[]) => React.ReactElement;

export interface Stories {
  [key: string]: Story;
}

export const storyWrapper = (stories: Stories) => {
  const wrappedStories: Stories = {};
  Object.keys(stories).forEach((key) => {
    wrappedStories[key] = () => (
      <>
        <Theme />
        <MemoryRouter>{stories[key](knobs)}</MemoryRouter>
      </>
    );
  });
  return wrappedStories;
};

export const storyBuilder = (stories: Stories, storyPath: string) => {
  const wrappedStories: Stories = storyWrapper(stories);
  const builtStories = storiesOf(storyPath, module);
  builtStories.addDecorator(withKnobs as any);
  Object.keys(wrappedStories).forEach((key) => {
    builtStories.add(key, wrappedStories[key]);
  });
};
