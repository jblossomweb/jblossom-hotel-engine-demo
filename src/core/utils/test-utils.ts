export const randomNumber = (min: number, max: number) =>
  Math.random() * (max - min) + min;

export const randomInteger = (min: number, max: number) =>
  Math.floor(randomNumber(min, max));

export const randomString = (length: number) =>
  Array(length)
    .fill(0)
    .map(() => Math.random().toString(36).charAt(2))
    .join('');
