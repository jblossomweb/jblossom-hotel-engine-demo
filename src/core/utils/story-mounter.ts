/* eslint-disable @typescript-eslint/no-explicit-any */
import './enzyme-adapter';
import { shallow, mount, ReactWrapper } from 'enzyme';
import { Stories, mockKnobs } from './story-builder';

export interface MountedStory {
  wrapper: ReactWrapper;
  component: ReactWrapper;
  props: any;
  elements: { [key: string]: ReactWrapper<any> };
}

const storyMounter = (
  stories: Stories,
  mainComponent: React.Component<any> | React.FC<any>,
  getElements?: (component: ReactWrapper) => MountedStory['elements'],
  mountShallow?: boolean
) => {
  const mountedStories: { [key: string]: MountedStory } = {};
  // tslint:disable-next-line: ban-types
  const mounter: Function = mountShallow ? shallow : mount;
  Object.keys(stories).forEach((key: string) => {
    const story = stories[key];
    const wrapper: ReactWrapper = mounter(story(mockKnobs));
    const component: ReactWrapper = wrapper.find(mainComponent);
    const props = component.props();
    const elements = getElements ? getElements(component) : {};
    mountedStories[key] = {
      wrapper,
      component,
      props,
      elements,
    };
  });
  return mountedStories;
};

export default storyMounter;
