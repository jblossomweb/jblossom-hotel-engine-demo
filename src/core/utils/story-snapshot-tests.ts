import renderer from 'react-test-renderer';
import { Story, Stories, mockKnobs } from './story-builder';

export const renderTree = (story: Story, classesWithRefs?: string[]) =>
  renderer
    .create(story(mockKnobs), {
      createNodeMock: (element) => {
        // optional workaround for known bug with react-test-renderer and refs
        if (classesWithRefs?.includes(element.props.className)) {
          return {
            scrollWidth: 0,
          };
        }
        return null;
      },
    })
    .toJSON();

const snapshotTests = (stories: Stories, classesWithRefs?: string[]) => {
  Object.keys(stories).forEach((key: string) => {
    const story = stories[key];
    describe(`${key} story`, () => {
      it('matches snapshot', () => {
        const tree = renderTree(story, classesWithRefs);
        expect(tree).toMatchSnapshot();
      });
    });
  });
};

export default snapshotTests;
