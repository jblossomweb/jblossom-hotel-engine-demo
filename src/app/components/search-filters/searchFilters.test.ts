import { Tag, Select } from 'antd';
import storyMounter from '../../../core/utils/story-mounter';
import snapshotTests from '../../../core/utils/story-snapshot-tests';
import { storyWrapper } from '../../../core/utils/story-builder';
import { makeStories } from './SearchFilters.stories';
import SearchFilters, { Props } from './SearchFilters';

const mockActions = {
  addFilteredLanguage: jest.fn(),
  removeFilteredLanguage: jest.fn(),
};

const stories = storyWrapper(makeStories(mockActions));
const mountedStories = storyMounter(stories, SearchFilters, (component) => ({
  tags: component.find(Tag),
  select: component.find(Select),
}));

describe('components/search-filters', () => {
  describe('snapshots', () => {
    snapshotTests(stories, ['ant-select-selection-search-mirror']);
  });

  describe('actions', () => {
    describe('click language pill x', () => {
      Object.keys(mountedStories).forEach((key) => {
        const story = mountedStories[key];
        const { loading, filteredLanguages }: Props = story.props;
        if (!loading && filteredLanguages?.length) {
          describe(`${key} story`, () => {
            filteredLanguages.forEach((language) => {
              describe(`click ${language} pill x`, () => {
                beforeEach(() => {
                  jest.clearAllMocks();
                });
                it(`calls props.removeFilteredLanguage('${language}')`, () => {
                  const { tags } = story.elements;
                  const tag = tags.filterWhere((node) =>
                    node.text().includes(language)
                  );
                  const xButton = tag
                    .find('[className="ant-tag-close-icon"]')
                    .first();
                  const { removeFilteredLanguage } = mockActions;
                  expect(removeFilteredLanguage).not.toHaveBeenCalled();
                  xButton.simulate('click');
                  expect(removeFilteredLanguage).toHaveBeenCalled();
                  expect(removeFilteredLanguage).toHaveBeenCalledWith(language);
                });
              });
            });
          });
        }
      });
    });

    describe('select a language from dropdown', () => {
      Object.keys(mountedStories).forEach((key) => {
        const story = mountedStories[key];
        const { loading, filteredLanguages }: Props = story.props;
        if (!loading) {
          describe(`${key} story`, () => {
            beforeEach(() => {
              jest.clearAllMocks();
            });
            ['foo', 'bar', 'buzz'].forEach((language) => {
              describe(`select ${language}`, () => {
                beforeEach(() => {
                  jest.clearAllMocks();
                });
                it(`calls props.addFilteredLanguage(${language})`, () => {
                  const { select } = story.elements;
                  const { onChange }: any = select.props();
                  const { addFilteredLanguage } = mockActions;
                  expect(addFilteredLanguage).not.toHaveBeenCalled();
                  onChange([...filteredLanguages, language]);
                  expect(addFilteredLanguage).toHaveBeenCalled();
                  expect(addFilteredLanguage).toHaveBeenCalledWith(language);
                });
              });
            });
          });
        }
      });
    });

    describe('onChange event fires but does not add (edge case)', () => {
      Object.keys(mountedStories).forEach((key) => {
        const story = mountedStories[key];
        const { loading, filteredLanguages }: Props = story.props;
        if (!loading) {
          describe(`${key} story`, () => {
            beforeEach(() => {
              jest.clearAllMocks();
            });
            it(`does not call props.addFilteredLanguage`, () => {
              const { select } = story.elements;
              const { onChange }: any = select.props();
              const { addFilteredLanguage } = mockActions;
              expect(addFilteredLanguage).not.toHaveBeenCalled();
              onChange(filteredLanguages); // same value as before
              expect(addFilteredLanguage).not.toHaveBeenCalled();
            });
          });
        }
      });
    });

    describe('onChange event fires, adds multiple (edge case)', () => {
      Object.keys(mountedStories).forEach((key) => {
        const story = mountedStories[key];
        const { loading, filteredLanguages }: Props = story.props;
        if (!loading) {
          describe(`${key} story`, () => {
            beforeEach(() => {
              jest.clearAllMocks();
            });
            it(`calls props.addFilteredLanguage for each new language`, () => {
              const { select } = story.elements;
              const { onChange }: any = select.props();
              const { addFilteredLanguage } = mockActions;
              expect(addFilteredLanguage).not.toHaveBeenCalled();
              onChange([...filteredLanguages, 'foo', 'bar', 'buzz']); // 3 new values
              expect(addFilteredLanguage).toHaveBeenCalledTimes(3);
            });
          });
        }
      });
    });
  });
});
