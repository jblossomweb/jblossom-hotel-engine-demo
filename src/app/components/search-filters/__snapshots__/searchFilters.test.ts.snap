// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/search-filters snapshots filter story matches snapshot 1`] = `
<div
  className="sc-AxjAm StDqN"
>
  <div
    className="ant-select ant-select-multiple ant-select-show-search"
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <div
      className="ant-select-selector"
      onMouseDown={[Function]}
    >
      <span
        onMouseDown={[Function]}
      >
        <span
          className="ant-tag"
          style={
            Object {
              "backgroundColor": undefined,
            }
          }
        >
          <span
            aria-label="code"
            className="anticon anticon-code"
            role="img"
          >
            <svg
              aria-hidden="true"
              className=""
              data-icon="code"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M516 673c0 4.4 3.4 8 7.5 8h185c4.1 0 7.5-3.6 7.5-8v-48c0-4.4-3.4-8-7.5-8h-185c-4.1 0-7.5 3.6-7.5 8v48zm-194.9 6.1l192-161c3.8-3.2 3.8-9.1 0-12.3l-192-160.9A7.95 7.95 0 00308 351v62.7c0 2.4 1 4.6 2.9 6.1L420.7 512l-109.8 92.2a8.1 8.1 0 00-2.9 6.1V673c0 6.8 7.9 10.5 13.1 6.1zM880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zm-40 728H184V184h656v656z"
              />
            </svg>
          </span>
           
          TypeScript
          <span
            aria-label="close"
            className="anticon anticon-close ant-tag-close-icon"
            onClick={[Function]}
            role="img"
            tabIndex={-1}
          >
            <svg
              aria-hidden="true"
              className=""
              data-icon="close"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 00203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
              />
            </svg>
          </span>
        </span>
      </span>
      <span
        className="ant-select-selection-search"
        style={
          Object {
            "width": 0,
          }
        }
      >
        <input
          aria-activedescendant="undefined_list_0"
          aria-autocomplete="list"
          aria-controls="undefined_list"
          aria-haspopup="listbox"
          aria-owns="undefined_list"
          autoComplete="off"
          className="ant-select-selection-search-input"
          disabled={false}
          onChange={[Function]}
          onCompositionEnd={[Function]}
          onCompositionStart={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onPaste={[Function]}
          readOnly={true}
          role="combobox"
          style={
            Object {
              "opacity": 0,
            }
          }
          unselectable="on"
          value=""
        />
        <span
          aria-hidden={true}
          className="ant-select-selection-search-mirror"
        >
          
           
        </span>
      </span>
    </div>
  </div>
</div>
`;

exports[`components/search-filters snapshots filters story matches snapshot 1`] = `
<div
  className="sc-AxjAm StDqN"
>
  <div
    className="ant-select ant-select-multiple ant-select-show-search"
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <div
      className="ant-select-selector"
      onMouseDown={[Function]}
    >
      <span
        onMouseDown={[Function]}
      >
        <span
          className="ant-tag"
          style={
            Object {
              "backgroundColor": undefined,
            }
          }
        >
          <span
            aria-label="code"
            className="anticon anticon-code"
            role="img"
          >
            <svg
              aria-hidden="true"
              className=""
              data-icon="code"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M516 673c0 4.4 3.4 8 7.5 8h185c4.1 0 7.5-3.6 7.5-8v-48c0-4.4-3.4-8-7.5-8h-185c-4.1 0-7.5 3.6-7.5 8v48zm-194.9 6.1l192-161c3.8-3.2 3.8-9.1 0-12.3l-192-160.9A7.95 7.95 0 00308 351v62.7c0 2.4 1 4.6 2.9 6.1L420.7 512l-109.8 92.2a8.1 8.1 0 00-2.9 6.1V673c0 6.8 7.9 10.5 13.1 6.1zM880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zm-40 728H184V184h656v656z"
              />
            </svg>
          </span>
           
          TypeScript
          <span
            aria-label="close"
            className="anticon anticon-close ant-tag-close-icon"
            onClick={[Function]}
            role="img"
            tabIndex={-1}
          >
            <svg
              aria-hidden="true"
              className=""
              data-icon="close"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 00203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
              />
            </svg>
          </span>
        </span>
      </span>
      <span
        onMouseDown={[Function]}
      >
        <span
          className="ant-tag"
          style={
            Object {
              "backgroundColor": undefined,
            }
          }
        >
          <span
            aria-label="code"
            className="anticon anticon-code"
            role="img"
          >
            <svg
              aria-hidden="true"
              className=""
              data-icon="code"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M516 673c0 4.4 3.4 8 7.5 8h185c4.1 0 7.5-3.6 7.5-8v-48c0-4.4-3.4-8-7.5-8h-185c-4.1 0-7.5 3.6-7.5 8v48zm-194.9 6.1l192-161c3.8-3.2 3.8-9.1 0-12.3l-192-160.9A7.95 7.95 0 00308 351v62.7c0 2.4 1 4.6 2.9 6.1L420.7 512l-109.8 92.2a8.1 8.1 0 00-2.9 6.1V673c0 6.8 7.9 10.5 13.1 6.1zM880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zm-40 728H184V184h656v656z"
              />
            </svg>
          </span>
           
          JavaScript
          <span
            aria-label="close"
            className="anticon anticon-close ant-tag-close-icon"
            onClick={[Function]}
            role="img"
            tabIndex={-1}
          >
            <svg
              aria-hidden="true"
              className=""
              data-icon="close"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 00203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
              />
            </svg>
          </span>
        </span>
      </span>
      <span
        className="ant-select-selection-search"
        style={
          Object {
            "width": 0,
          }
        }
      >
        <input
          aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
          aria-autocomplete="list"
          aria-controls="rc_select_TEST_OR_SSR_list"
          aria-haspopup="listbox"
          aria-owns="rc_select_TEST_OR_SSR_list"
          autoComplete="off"
          className="ant-select-selection-search-input"
          disabled={false}
          id="rc_select_TEST_OR_SSR"
          onChange={[Function]}
          onCompositionEnd={[Function]}
          onCompositionStart={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onPaste={[Function]}
          readOnly={true}
          role="combobox"
          style={
            Object {
              "opacity": 0,
            }
          }
          unselectable="on"
          value=""
        />
        <span
          aria-hidden={true}
          className="ant-select-selection-search-mirror"
        >
          
           
        </span>
      </span>
    </div>
  </div>
</div>
`;

exports[`components/search-filters snapshots loading story matches snapshot 1`] = `
<div
  className="sc-AxjAm StDqN"
>
  <div
    className="ant-select ant-select-multiple ant-select-disabled ant-select-show-search"
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <div
      className="ant-select-selector"
      onMouseDown={[Function]}
    >
      <span
        className="ant-select-selection-search"
        style={
          Object {
            "width": 0,
          }
        }
      >
        <input
          aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
          aria-autocomplete="list"
          aria-controls="rc_select_TEST_OR_SSR_list"
          aria-expanded={false}
          aria-haspopup="listbox"
          aria-owns="rc_select_TEST_OR_SSR_list"
          autoComplete="off"
          className="ant-select-selection-search-input"
          disabled={true}
          id="rc_select_TEST_OR_SSR"
          onChange={[Function]}
          onCompositionEnd={[Function]}
          onCompositionStart={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onPaste={[Function]}
          readOnly={true}
          role="combobox"
          style={
            Object {
              "opacity": 0,
            }
          }
          unselectable="on"
          value=""
        />
        <span
          aria-hidden={true}
          className="ant-select-selection-search-mirror"
        >
          
           
        </span>
      </span>
      <span
        className="ant-select-selection-placeholder"
      >
        <span
          aria-label="code"
          className="anticon anticon-code"
          role="img"
        >
          <svg
            aria-hidden="true"
            className=""
            data-icon="code"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M516 673c0 4.4 3.4 8 7.5 8h185c4.1 0 7.5-3.6 7.5-8v-48c0-4.4-3.4-8-7.5-8h-185c-4.1 0-7.5 3.6-7.5 8v48zm-194.9 6.1l192-161c3.8-3.2 3.8-9.1 0-12.3l-192-160.9A7.95 7.95 0 00308 351v62.7c0 2.4 1 4.6 2.9 6.1L420.7 512l-109.8 92.2a8.1 8.1 0 00-2.9 6.1V673c0 6.8 7.9 10.5 13.1 6.1zM880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zm-40 728H184V184h656v656z"
            />
          </svg>
        </span>
         language
      </span>
    </div>
  </div>
</div>
`;

exports[`components/search-filters snapshots sample story matches snapshot 1`] = `
<div
  className="sc-AxjAm StDqN"
>
  <div
    className="ant-select ant-select-multiple ant-select-show-search"
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <div
      className="ant-select-selector"
      onMouseDown={[Function]}
    >
      <span
        className="ant-select-selection-search"
        style={
          Object {
            "width": 0,
          }
        }
      >
        <input
          aria-activedescendant="undefined_list_0"
          aria-autocomplete="list"
          aria-controls="undefined_list"
          aria-haspopup="listbox"
          aria-owns="undefined_list"
          autoComplete="off"
          className="ant-select-selection-search-input"
          disabled={false}
          onChange={[Function]}
          onCompositionEnd={[Function]}
          onCompositionStart={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onPaste={[Function]}
          readOnly={true}
          role="combobox"
          style={
            Object {
              "opacity": 0,
            }
          }
          unselectable="on"
          value=""
        />
        <span
          aria-hidden={true}
          className="ant-select-selection-search-mirror"
        >
          
           
        </span>
      </span>
      <span
        className="ant-select-selection-placeholder"
      >
        <span
          aria-label="code"
          className="anticon anticon-code"
          role="img"
        >
          <svg
            aria-hidden="true"
            className=""
            data-icon="code"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M516 673c0 4.4 3.4 8 7.5 8h185c4.1 0 7.5-3.6 7.5-8v-48c0-4.4-3.4-8-7.5-8h-185c-4.1 0-7.5 3.6-7.5 8v48zm-194.9 6.1l192-161c3.8-3.2 3.8-9.1 0-12.3l-192-160.9A7.95 7.95 0 00308 351v62.7c0 2.4 1 4.6 2.9 6.1L420.7 512l-109.8 92.2a8.1 8.1 0 00-2.9 6.1V673c0 6.8 7.9 10.5 13.1 6.1zM880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zm-40 728H184V184h656v656z"
            />
          </svg>
        </span>
         language
      </span>
    </div>
  </div>
</div>
`;
